packages/react/src/hooks/useTrackToggle.ts

```diff
@@ -65,7 +65,7 @@ export function useTrackToggle<T extends ToggleSource>({
   const clickHandler: React.MouseEventHandler<HTMLButtonElement> = React.useCallback(
     (evt) => {
       userInteractionRef.current = true;
-      toggle().finally(() => (userInteractionRef.current = false));
+      toggle().catch(() => (userInteractionRef.current = false));
```

Comment:
we're only interested in resetting this on failures, as otherwise it will be reset above in the `useEffect`. 

We don't want to use `finally` here as it introduces a race around the emitted value of the `userInitiated` boolean